common/Resources/ti.internal/extensions/process.js

```diff
@@ -46,14 +46,87 @@ function standardizeArch(original) {
 	}
 }
 
+const listeners = {};
 const process = {
 	arch: standardizeArch(Ti.Platform.architecture),
+	noDeprecation: false,
+	throwDeprecation: false,
+	traceDeprecation: false,
+	pid: 0,
 	cwd: function () {
 		return __dirname;
 	},
 	// FIXME: Should we try and adopt 'windowsphone'/'windowsstore' to 'win32'?
 	// FIXME: Should we try and adopt 'ipad'/'iphone' to 'darwin'? or 'ios'?
-	platform: Ti.Platform.osname
+	platform: Ti.Platform.osname,
+	on: function (eventName, callback) {
+		const eventListeners = listeners[eventName] || [];
+		eventListeners.push(callback);
+		listeners[eventName] = eventListeners;
+		return this;
+	},
+	// TODO: Add #once which is like #on, but should get wrapped to remove itself before getting fired
```

Comment:
We also might want to move the event stuff into a separate module so it can be reused.

Reply:
agreed. As I go I am eventually extracting the code out to it's proper place, but didn't want to go nuts writing every other module as these PRs are large enough. But, yeah we need to define EventEmitter eventually.

Reply:
Alright, sounds good!